src/web/hooks/use-request-modal.ts

```diff
@@ -1,8 +1,19 @@
+import { useOpenAiGlobal } from "./use-openai-global.js";
+
 /**
  * Triggers a modal containing the widget rendered in display mode "modal"
  */
 export function useRequestModal() {
-  return (options: { title: string }) => {
+  const displayMode = useOpenAiGlobal("displayMode");
+  const isOpen = displayMode === "modal";
+
+  const open = (options: {
+    title?: string;
+    params?: Record<string, unknown>;
+    anchor?: { top?: number; left?: number; width?: number; height?: number };
```

Comment:
To avoid duplication, export type from `types.ts` like it was done for `CallToolArgs` or `CallToolResponse` and use it here :)